File: Question 0101 - Symmetric Tree/src/Solution.java
class Solution
{
    private boolean solve(TreeNode root1, TreeNode root2)
    {
        if(root1 == null && root2 == null) return true;
        if(root1 != null && root2 == null) return false;
        if(root1 == null && root2 != null) return false;

        boolean left = solve(root1.left, root2.right);
        boolean right = solve(root1.right, root2.left);

        return(root1.val == root2.val && left && right);
    }
    public boolean isSymmetric(TreeNode root)
    {
        if(root == null) return true;
        return solve(root.left, root.right);
    }
}

File: Question 0101 - Symmetric Tree/src/Solution.cpp
#include <bits/stdc++.h>
using namespace std;

struct TreeNode
{
    int val;
    TreeNode *left;
    TreeNode *right;
    TreeNode() : val(0), left(nullptr), right(nullptr) {}
    TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
    TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
};

class Solution
{
private:
    bool solve(TreeNode *t1, TreeNode *t2)
    {
        if(!t1 && !t2)
            return true;

        if((!t1 && t2) || (t1 && !t2))
            return false;

        return solve(t1->left, t2->right) && solve(t1->right, t2->left) && t1->val == t2->val;
    }

public:
    bool isSymmetric(TreeNode *root)
    {
        return solve(root->left, root->right);
    }
};